wp-includes/general-template.php

```diff
@@ -101,13 +101,16 @@ function wp_add_error_template_no_robots() {
 }
 
 /**
- * Hide the admin bar if serving the offline template.
+ * Ensure current user is unset when serving an error page (either offline or server error).
  *
- * @since 0.2
+ * This is important so that what the service worker initially precaches for the user when first accessing the site
+ * will persist even after they have authenticated.
+ *
+ * @since 0.5
  */
-function wp_hide_admin_bar_offline() {
-	if ( is_offline() ) {
-		show_admin_bar( false );
+function wp_unset_current_user_error_template() {
+	if ( is_offline() || is_500() ) {
+		wp_set_current_user( 0 );
 	}
 }
```

Comment:
Can we maybe bake this somehow as a method into `WP_Service_Worker_Navigation_Routing_Component` instead? It's a bit hard to digest what it relates to from here and why this is being done, and that class mainly deals with these two templates. It also contains the other change (remove user-specific affix for revision) that makes this here necessary.

Reply:
Humm. Thinking about this. The `WP_Service_Worker_Navigation_Routing_Component` class is about referencing the URL to the offline/500 page; it isn't involved in actually serving it. 

WordPress would serve these templates via `wp-includes/template-loader.php`, though at the moment the PWA plugin implements that via the `template_include` filter. What we need to do here is ensure that when that `template_include` logic is run to include the offline/500 template, the user is unset. The earliest point we can check `is_offline()` or `is_500()` is at the `parse_query` action, and that is when here `wp_unset_current_user_error_template()` is called.

So I think this is pretty good where it is here actually. 

I've added some tests in f0e5984.

Reply:
Hmm, I see what you're saying related to serving the request. On the other hand, unsetting the user is necessary because of the logic in `WP_Service_Worker_Navigation_Routing_Component`. If it didn't provide the 500/offline templates to the service-worker for precaching, `wp_unauthenticate_error_template_requests` wouldn't be necessary.

Reply:
It wouldn't be necessary if `WP_Service_Worker_Navigation_Routing_Component` were able to construct service worker JS in a way that would prevent cookies from being sent in the request to precache the offline/500 page.

Reply:
Makes sense. Hence my other general comment - maybe that is not technically possible, but if it was, I think it'd be a nicer solution.